File: Xenon/Xenon/Explosion.ts
import CActor = require("Actor");

class CExplosion extends CActor {
    public CExplosion() {
    }

    activate() {
        if (!this.isActive()) {
            //	m_timer.start();
        }

        return super.activate();
    }

    //update(controls: Controls, gametime: GameTime) {
    //    this.gameTime = gametime;
    //    //this.m_position += this.m_velocity;

    //    if (super.animate(AnimationMode.ANIMATE_ONESHOT)) {
    //        super.kill();
    //    }

    //    return true;
    //}

    onLeavingScreen() {
        super.kill();
    }

}
export = CExplosion;

File: Xenon-Original_C++_Code/xenon/source/explosion.cpp
//-------------------------------------------------------------
//
// Class:	CExplosion
//
// Author:	John M Phillips
//
// Started:	06/05/00
//
// Base:	CAlien
//
// Derived:	CSmallExplosion
//			CMediumExplosion
//			CBigExplosion
//
//-------------------------------------------------------------

#include "game.h"

//-------------------------------------------------------------

CExplosion::CExplosion()
{
}

//-------------------------------------------------------------

CExplosion::~CExplosion()
{
//	CActor::~CActor();
}

//-------------------------------------------------------------

bool CExplosion::activate()
{
	if (!isActive()) {
		m_timer.start();
		}

	return CActor::activate();
}

//-------------------------------------------------------------

bool CExplosion::update(Controls *controls)
{
	m_position += m_velocity;

	if (animate(ANIMATE_ONESHOT))
		kill();
	
	return true;
}

//-------------------------------------------------------------

void CExplosion::onLeavingScreen()
{
	kill();
}

//-------------------------------------------------------------

bool CSmallExplosion::activate()
{
	CGameState::playSample(SAMPLE_SMALL_EXPLOSION,getPosition().getX());

	return CExplosion::activate();
}

//-------------------------------------------------------------

bool CMediumExplosion::activate()
{
	CGameState::playSample(SAMPLE_MEDIUM_EXPLOSION,getPosition().getX());

	return CExplosion::activate();
}

//-------------------------------------------------------------

bool CBigExplosion::activate()
{
	CGameState::playSample(SAMPLE_BIG_EXPLOSION,getPosition().getX());

	return CExplosion::activate();
}

//-------------------------------------------------------------
